### PR TITLE
fix: update angular to beta.17

### PR DESCRIPTION
**Pull request by jelbourn**
R: @hansl @kara

### CODE REVIEW

File: src/components/sidenav/sidenav.spec.ts [hunk 1]
```diff
@@ -77,6 +77,7 @@ export function main() {
         xhr.get('./components/sidenav/sidenav.css').then((css) => { style = css; })
       ]).catch((err: any) => {
         console.error(err);
+        console.error(err);
```
Comment (kara) [inline src/components/sidenav/sidenav.spec.ts]: Duplication by mistake?

File: src/demo-app/input/input-demo.html [hunk 2]
```diff
@@ -145,7 +145,7 @@
         <button (click)="addABunch(1000)">Add 1000</button>
       </td>
     </thead>
-    <tr *ngFor="#item of items; #i = index">
+    <tr *ngFor="let item of items; #i = index">
```
Comment (kara) [inline src/demo-app/input/input-demo.html]: Looks like you need to replace the `#i` here with `let i`?

File: src/components/sidenav/sidenav.spec.ts (hunk 1 above)
Comment (jelbourn) [inline src/components/sidenav/sidenav.spec.ts]: Fixed.

File: src/demo-app/input/input-demo.html (hunk 2 above)
Comment (jelbourn) [inline src/demo-app/input/input-demo.html]: Fixed.

**Pull request merged**